types/base.d.ts

```diff
@@ -120,5 +120,13 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 	context: any;
 }
 
+export interface Macro<T, I, E> {
+	(t: T, input: I, expected: E): void;
+	title? (providedTitle: string, input: I, expected: E): string;
+}
+export type Macros<T, I, E> = Macro<T, I, E> | Macro<T, I, E>[];
+
 export function test(name: string, run: ContextualTest): void;
 export function test(run: ContextualTest): void;
+export function test<I, E> (name: string, run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
+export function test<I, E> (run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
```

Comment:
Zero or more arguments can be provided to a macro. Rather than specifying `input` and `expected` I think this should just be `...args: any[]` (not 100% certain on the syntax).